src/Silex/ArgumentResolver.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Silex framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Silex;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolverInterface;
+
+/**
+ * HttpKernel Argument Resolver for Silex.
+ *
+ * @author Romain Neutron <imprec@gmail.com>
+ */
+class ArgumentResolver implements ArgumentResolverInterface
+{
+    private $ar;
+    private $app;
+
+    public function __construct(ArgumentResolverInterface $ar)
+    {
+        $this->ar = $ar;
+    }
+
+    public function setApplication(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    public function getArguments(Request $request, $controller)
+    {
+        if (null === $this->app) {
+            throw new \LogicException('You should inject Silex Application prior to call ArgumentResolver::getArguments');
```

Comment:
woops, I agree ! I started the implemenation by *extending* Symfony's ArgumentResolver and signature is quite big. I finally went to decoration because the class is final.

sorry :) Gonna update this